unittest/ota_voice_test.sh

```diff
@@ -181,7 +183,11 @@ usb_lsb_upper=$(echo ${usb_lsb} | awk '{print toupper($0)}')
 run_rigctl "\\set_mode PKT${usb_lsb_upper} 0" $model
 run_rigctl "\\set_freq ${freq_Hz}" $model
 run_rigctl "\\set_ptt 1" $model
-aplay --device="plughw:CARD=CODEC,DEV=0" -f S16_LE tx.raw 2>/dev/null
+if [ `uname` == "Darwin" ]; then
+    play -t raw -b 16 -c 1 -r 8000 -e signed-integer --endian little tx.raw 
+else
+    aplay --device="plughw:CARD=CODEC,DEV=0" -f S16_LE tx.raw 2>/dev/null
+fi
```

Comment:
You might be able to use `play -t -.s16 -r 8000 -c 1 .....`

Yes the `--device="plughw:CARD=CODEC,DEV=0"` selects the IC7200 USB device by name so I don't have to set it as the default sound card.  They may be an equivalent `sox` method on macOS.

Reply:
That command didn't work, unfortunately, but I did find out how to override the output audio device. Unfortunately there seem to be limits, as evidenced here:

```
HT-TM05:unittest mooneer$ sox -V6 -n -t coreaudio junkname
sox:      SoX v
time:     Dec  1 2020 00:54:51
issue:    macosx
uname:    Darwin Mooneers-16-MBP.home.lifeafterking.org 20.5.0 Darwin Kernel Version 20.5.0: Sat May  8 05:10:33 PDT 2021; root:xnu-7195.121.3~9/RELEASE_X86_64 x86_64
compiler: gcc Apple LLVM 12.0.0 (clang-1200.0.32.27)
arch:     1288 48 88 L 
sox INFO nulfile: sample rate not specified; using 48000

Input File     : '' (null)
Channels       : 1
Sample Rate    : 48000
Precision      : 32-bit

sox INFO coreaudio: Found Audio Device "Background Music"

sox INFO coreaudio: Found Audio Device "Background Music"

sox INFO coreaudio: Found Audio Device "MacBook Pro Micr"

sox INFO coreaudio: Found Audio Device "MacBook Pro Spea"

sox INFO coreaudio: Found Audio Device "VB-Cable"

sox INFO coreaudio: Found Audio Device "VB-Cable"

sox INFO coreaudio: Found Audio Device "VB-Cable"

sox INFO coreaudio: Found Audio Device "ZoomAudi"

sox FAIL formats: can't open output file `junkname': can not open audio device
```

(I should be seeing "VB-Cable", "VB-Cable A" and "VB-Cable B". The latter is the one my SDR is normally set to for TX.)

Reply:
Oh dear, close but not quite there :thinking:

Reply:
Not sure there's much that can be done about that, unfortunately. That seems like a sox bug that they'll need to fix. Though perhaps I can run a MacPorts selfupdate and see if there's a new version available?

Reply:
I'm actually using an old Linux laptop as a server to run the tests.  That way when the cron stuff is set up I can leave it connected to the radio and running while my main development laptop get moved about through my working day.

Running `ssh -X` I can display the plots, and even replay wave files on my local machine using:
```
scp bear.lan:codec2/unittest/rx_freedv.wav /dev/stdout | aplay -f S16_LE
```